lib/actions/preload.coffee

```diff
@@ -18,24 +18,52 @@ dockerUtils = require('../utils/docker')
 
 LATEST = 'latest'
 
+deviceTypes = undefined
+
+getDeviceTypes = ->
+	Bluebird = require('bluebird')
+	if deviceTypes != undefined
+		return Bluebird.resolve(deviceTypes)
+	resin = require('resin-sdk').fromSharedOptions()
+	resin.models.config.getDeviceTypes()
+	.then (dt) ->
+		deviceTypes = dt
+		return deviceTypes
+
+getDeviceTypesWithSameArch = (deviceType) ->
+	_ = require('lodash')
+
+	getDeviceTypes()
+	.then (deviceTypes) ->
+		deviceType = _.find deviceTypes, (dt) ->
```

Comment:
deviceType changes type here, which is pretty confusing. Can we call the parameter `deviceTypeSlug` instead, and have two separate variables?